pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -685,13 +687,17 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 
 					Context("and the InfraMachine does not exist", func() {
 						It("should update the synchronized condition on the MAPI machine to False", func() {
+							// TODO: Revert this to a useful error message.
```

Comment:
Because we changed `fetchCAPIInfraResources` to return a Nil `InfraMachine` rather than error.

We changed this because we must delete the inframachine first - then the capi machine's deletion is unblocked. 

This needs more thought at a later date, but works and is tested on cluster for now.